### PR TITLE
UCP/RMA: Implementation of ucp_ep_flush

### PR DESCRIPTION
**Pull request by brminich**
- ucp_ep_flush implemented
- several tests for ucp_ep_flush added
- uct_empty_function_return_unsupported  renamed to
  ucs_empty_function_return_unsupported
- ucs_empty_function_return_inprogress added
- removed excessive UCP_RKEY_LOOKUP call in ucp_get_nbi

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/997/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1608/ for details.

**Comment by brminich** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1000/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1611/ for details.

File: src/ucp/rma/basic_rma.c [hunk 1]
```diff
@@ -395,3 +395,20 @@ ucs_status_t ucp_worker_flush(ucp_worker_h worker)
...
+
+ucs_status_t ucp_ep_flush(ucp_ep_h ep)
+{
+    ucs_status_t status;
+
+    /* TODO all uct endpoints need to be flushed. Currenlty ucp endpoint
+     * supports just one uct endpoint. */
+    for (;;) {
+        status = uct_ep_flush(ep->uct_ep);
+        if (status != UCS_INPROGRESS && status != UCS_ERR_NO_RESOURCE) {
```
Comment (yosefe) [inline src/ucp/rma/basic_rma.c]: pls use () for each expression

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -673,6 +673,23 @@ void ucp_ep_destroy(ucp_ep_h ep);
 
 
 /**
+ * @ingroup UCP_ENDPOINT
+ *
+ * @brief Flush outstanding one-sided operations on the @ref ucp_ep_h "endpoint"
+ *
+ * This routine flushes all outstanding one-sided communications on the
+ * @ref ucp_ep_h "endpoint".  Communication operations issued on the @a ep
+ * prior to this call will have been completed both at the origin and at the
```
Comment (yosefe) [inline src/ucp/api/ucp.h]: "will have" -> "are"

**Comment by yosefe** [comment]
looks good besides minor comments

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1617/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1005/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
:+1:

**Comment by shamisp** [comment]
@brminich next time I would suggest to create a separate pr for each one of the issues. This is particularly relevant for a separation of new features and bugfixes

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1630/ for details.

**Pull request merged**